### PR TITLE
feat: CP support Image closeIcon

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- #46889

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support Image closeIcon |
| 🇨🇳 Chinese | feat: CP support Image closeIcon |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/config-provider/context.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import * as React from 'react';
+import type { PreviewProps } from 'rc-image/es/Preview';
```
Comment (MadCcc) [inline components/config-provider/context.ts:2]: 这个还是从 ImageProps 里取比较好

Comment (li-jia-nan) [reply components/config-provider/context.ts:2]: ImageProps 里没暴露出来

File: components/image/index.tsx [hunk 2]
```diff
@@ -68,6 +68,7 @@ const Image: CompositionImage<ImageProps> = (props) => {
       transitionName: getTransitionName(rootPrefixCls, 'zoom', _preview.transitionName),
       maskTransitionName: getTransitionName(rootPrefixCls, 'fade', _preview.maskTransitionName),
       zIndex,
+      closeIcon: _preview.closeIcon ?? image?.preview?.closeIcon ?? icons.close,
```
Comment (MadCcc) [inline components/image/index.tsx:71]: 1. closeIcon 从 _preview 解构一下
2. image.preview.closeIcon 需要加入 useMemo 的 deps
3. icons.close 不需要透出

Comment (li-jia-nan) [reply components/image/index.tsx]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**